### PR TITLE
Independent Y axis scale w/ band connections

### LINKED ISSUE #286: Independent y-axis scale
We need to allow users to use independent scales of y-axis between metadata visualizations. Currently, we are using fixed and shared y-axis across HiGlass and metadata visualizations. So, visual elements (e.g., cells, bars, texts) in each row correspond to the same sample. 

However, it is sometimes useful to use independent y-axis scale between visualizations. One good example would be in dendrograms. By having an independent scale, we can show dendrogram every time regardless of the arrangement state (i.e., `selectedRows`). Currently, we are hiding dendrograms if arrangements of samples in HiGlass does not match to the arrangements of dendrograms.

**What to update**

- Add an option to use independent y-axis scale in the grammar.
- Support this option interactively in HiGlassMeta.
- Show line or band connections between visualizations to visually couple between them (connecting the corresponding rows).
   - We can provide a track-level option for using a different way of linking samples (straight, curved, no lines, ...)
- Possibly, we can support interactive support for this.

Refer to Domino for visual linkings (https://caleydo.org/publications/2014_infovis_domino/).

### LINKED ISSUE #292: Hide track titles upon mouse hover on the track
Track titles disallow mouse hovering on bars that are positioned on the top, which make it unable to see detailed information with tooltips.

### PR DESCRIPTION
**Pull request by sehilyi**
Screen Shot 2020-07-28 at 9 20 01 PM

Sorry, this is such a large update.

**Updates**
- Allow using independent y axis scale using `resolveYScale` and `sort` options (Fixes #286)
- Added line/band connections between tracks whey they use different y-axis scales.
- Added linking tracks with background color upon mouse hover.
- Added `width` options that set the initial size of vertical tracks along x-axis.

**Other minor updates**
- Hide titles upon mouse hover on the corresponding track (Fixes #292)
- Added an `onRemoveAllSort` API in `HiGlassMeta` for removing all sort options outside of the library.
- Temporally changed the row highlighting method, from reverse highlighting to regular highlighting, i.e., highlight the selected rows. 
   - The reason for this update was to use a consistent linking methods between tracks (i.e., same background color). Tested with reverse highlighting for every tracks but was quite annoying to see.

**Possible improvements in the next PR**
- Use `index` instead of `range` for interactively highlighting rows from quantitative bar charts.
- Do not need to capture all highlighting history for undo/redo (Perhaps, add `observe`/`ignore` options for each interaction).
- Use edge-bundling or other enhanced techniques for line/band representations.
- Turn on/off option for highlighting upon mouse hover since this can be annoying to see.

### CODE REVIEW

File: src/demo/CistromeExplorer.js [hunk 1]
```diff
@@ -5,253 +5,11 @@ import { HiGlassMeta } from '../index.js';
...
-import hgDemoViewConfig6 from '../viewconfigs/horizontal-multivec-6.json';
-import hgDemoViewConfig7 from '../viewconfigs/horizontal-multivec-7.json';
-import hgDemoViewConfig8 from '../viewconfigs/horizontal-multivec-8.json';
-import hgDemoViewConfig9 from '../viewconfigs/horizontal-multivec-9.json';
-import hgDemoViewConfig10 from '../viewconfigs/horizontal-multivec-10.json';
-import hgDemoViewConfigApril2020 from '../viewconfigs/meeting-2020-04-29.json';
-
-import './CistromeExplorer.scss';
 import { diffViewOptions } from '../utils/view-history';
+import { demos } from './demo';
```
Comment (keller-mark) [inline src/demo/CistromeExplorer.js:9]: Thanks for moving this, it has become a large chunk of code!

File: src/demo/CistromeExplorer.js [hunk 2]
```diff
@@ -325,7 +83,14 @@ export default function CistromeExplorer() {
...
     return (
-        <div className="cistrome-explorer">
+        <div className="cistrome-explorer"
+            onKeyDown={e => {
+                if(
+                    (e.key === 'Esc' || e.key === 'Escape') && isSettingVisible
+                ) {
+                    setIsSettingVisible(false);
+                }
+            }}>
```
Comment (keller-mark) [inline src/demo/CistromeExplorer.js:93]: It seems like the Escape key only works after I have clicked the `<select/>` dropdown element. Would you be able to move this to an event listener on the window?

```js
useEffect(() => {
    function closeSidebar(e) {
        if((e.key === 'Esc' || e.key === 'Escape') && isSettingVisible) {
            setIsSettingVisible(false);
        }
    }
    window.addEventListener("keydown", closeSidebar);
    return () => window.removeEventListener("keydown", closeSidebar);
}, [isSettingVisible]);
```

**Comment by keller-mark** [comment]
The screenshot really doesn't do this justice. It is also really great that the dendrogram can always be visible now!

File: src/demo/CistromeExplorer.js (hunk 2 above)
Comment (sehilyi) [reply src/demo/CistromeExplorer.js:93]: Thanks for suggesting this! This is much consistent. Included the Toolkit view here as well.

**Pull request merged**